### PR TITLE
Feature: Add Empty State Display for Tables

### LINKED ISSUE #1991: Empty state V1 for tables
## Current behavior

There is no empty state, we display a blank page.

image

## Desired behavior

We should add this empty state on both **Standard objects** and **Custom objects**

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=12639-96903&mode=design&t=uMog7Tqon1L3Ogmq-4

### PR DESCRIPTION
**Pull request by tmahabir**
Resolves #1991 

Added empty state display for when there are no records for Standard objects and Custom objects.
Here's a visual of the change:

https://github.com/twentyhq/twenty/assets/55632829/99aca1e2-de35-466b-b283-457f81d9b88f

### CODE REVIEW

File: front/src/modules/object-record/record-table/components/RecordTable.tsx [hunk 1]
```diff
@@ -90,9 +125,34 @@ export const RecordTable = ({
 }: RecordTableProps) => {
   const tableBodyRef = useRef<HTMLDivElement>(null);
 
-  const { resetTableRowSelection, setRowSelectedState } = useRecordTable({
+  const tableRowIds = useRecoilValue(tableRowIdsState);
```
Comment (thaisguigon) [inline front/src/modules/object-record/record-table/components/RecordTable.tsx:128]: Here we could use `numberOfTableRowsState` instead.

File: front/src/modules/object-record/record-table/components/RecordTable.tsx [hunk 2]
```diff
@@ -90,9 +125,34 @@ export const RecordTable = ({
...
+    },
+  );
+
+  const { createOneRecord: createOneObject } = useCreateOneRecord({
+    objectNameSingular,
+  });
+
+  const handleAddButtonClick = async () => {
+    await createOneObject?.({});
+  };
```
Comment (thaisguigon) [inline front/src/modules/object-record/record-table/components/RecordTable.tsx:154]: The `createRecord` prop already does this, let's use the prop instead!

**Review by thaisguigon — CHANGES_REQUESTED** [review]
Hi @tmahabir, thank you for your contribution!

I rebased your PR as some important fixes were merged recently. Left a few minor comments, otherwise LGTM!

File: front/src/modules/object-record/record-table/components/RecordTable.tsx (hunk 1 above)
Comment (tmahabir) [reply front/src/modules/object-record/record-table/components/RecordTable.tsx]: Done

File: front/src/modules/object-record/record-table/components/RecordTable.tsx (hunk 2 above)
Comment (tmahabir) [reply front/src/modules/object-record/record-table/components/RecordTable.tsx]: Done

**Comment by tmahabir** [comment]
Hi @thaisguigon ! Thank you for the comments, I've addressed them in my most recent commits!

**Review by thaisguigon — APPROVED** [review]
Approved, thank you!

**Comment by charlesBochet** [comment]
Merging it!

**Pull request merged**